crates/miden-objects/src/account/delta/mod.rs

```diff
@@ -238,10 +239,21 @@ impl AccountDelta {
     /// generally). Including `num_changed_entries` disambiguates this situation, by ensuring
     /// that the delta commitment is different when, e.g. 1) a non-fungible asset and one key-value
     /// pair have changed and 2) when two key-value pairs have changed.
-    pub fn commitment(&self) -> Word {
+    pub fn to_commitment(&self) -> Word {
+        <Self as SequentialCommit>::to_commitment(self)
+    }
```

Comment:
It's a bit odd that we have two `to_commitment`s on `AccountDelta`, and I assume the reason for having these is to keep the documentation here and not have to reimplement `to_commitment`? Not a strong opinion, but I think I would remove this method and instead override the implementation of `SequentialCommit::to_commitment` and move the docs there. Basically, I prefer having one way of doing something.

I feel like this sets a precedence for the use of `SequentialCommit` in the codebase, and whatever we do here should also apply to other implementations of `SequentialCommit`.

Reply:
Makes sense. We should then apply this approach as well when we implement `SequentialCommit` for other types I think.